src/Validation/gen/Parsers/ValidationsGenerator.AttributeParser.cs

```diff
@@ -12,20 +13,121 @@ namespace Microsoft.Extensions.Validation;
+        GeneratorSyntaxContext context,
+        GeneratorSettings settings,
+        CancellationToken cancellationToken)
+    {
+        if (context.Node is not (ClassDeclarationSyntax or RecordDeclarationSyntax))
+        {
+            return [];
+        }
+
+        var typeSymbol = context.SemanticModel.GetDeclaredSymbol(context.Node, cancellationToken) as ITypeSymbol;
+        if (typeSymbol == null)
+        {
+            return [];
+        }
+
+        var validatableTypeAttributeName = $"{settings.RootNamespace}.ValidatableTypeAttribute";
+        // Check if the type has a ValidatableTypeAttribute (auto-generated)
+        if (!HasValidatableTypeAttribute(typeSymbol, validatableTypeAttributeName))
+        {
+            return [];
+        }
+
+        var validatableTypes = new HashSet<ValidatableType>(ValidatableTypeComparer.Instance);
+        List<ITypeSymbol> visitedTypes = [];
+        var wellKnownTypes = WellKnownTypes.GetOrCreate(context.SemanticModel.Compilation);
+
+        if (TryExtractValidatableType(typeSymbol, wellKnownTypes, ref validatableTypes, ref visitedTypes))
+        {
+            return [.. validatableTypes];
+        }
+        return [];
+    }
+
+    private static bool IsValidatableTypeAttribute(AttributeSyntax attribute)
+    {
+        var name = attribute.Name.ToString();
+        return name is "ValidatableType" or "ValidatableTypeAttribute" ||
+               name.EndsWith(".ValidatableType", StringComparison.Ordinal) ||
+               name.EndsWith(".ValidatableTypeAttribute", StringComparison.Ordinal);
+    }
+
+    private static bool HasValidatableTypeAttribute(ITypeSymbol typeSymbol, string validatableTypeAttributeName)
+    {
+        foreach (var attr in typeSymbol.GetAttributes())
+        {
+            var attributeClass = attr.AttributeClass;
+            if (attributeClass is null)
+            {
+                continue;
+            }
+
+            var name = attributeClass.Name;
+            var namespaceName = attributeClass.ContainingNamespace.ToDisplayString();
+            var fullName = $"{namespaceName}.{name}";
+            if (!string.Equals(fullName, validatableTypeAttributeName, StringComparison.Ordinal))
+            {
+                continue;
+            }
+
+            // Check for auto-generated attribute (any namespace)
+            if (name == validatableTypeAttributeName)
+            {
+                // Additional check: ensure it's marked with [Embedded] to confirm it's auto-generated
+                foreach (var embeddedAttr in attributeClass.GetAttributes())
+                {
+                    if (embeddedAttr.AttributeClass?.ToDisplayString() == "Microsoft.CodeAnalysis.EmbeddedAttribute")
+                    {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
```

Comment:
This I think is fragile, I'm not sure if the best way would be to use the generatorsettings to try and get the generated validatableattribute symbol and compare against that, or to compute it with a syntax provider and combine that instead of the settings (I'm not sure if we can get the symbol for the postinitializationoutput at all)